### ISSUE TITLE
NVIDIA nv0 training stalls beyond 60s on quantized layer/pool model with integer arithmetic

### ISSUE THREAD

**Issue by nm-z** [nm-z/recipe-dev#314]
Public `recipe.train().optimizer(adamw).lr(0.0001).seed(46945512480253).epochs(1).int(8).run()` on `backend:nv0` exceeds the 60-second threshold for one epoch on `data/numeric/sample_subfolders` using `recipe.model().layer(4).elu().norm(batch).iq(1).m .pool(2).exp().norm(batch).iq(1).m .loss(huber)`. The earliest failure is the nv0 quantized native-artifact stall during the first epoch, triggered by `.int(8)` integer arithmetic combined with batch-normalized quantized layer and pool nodes; no existing nv0 issue covers this layer/pool path (residual/attention/GRU issues do not match). Measured 60s is materially slower than the small-row numeric workload justifies (expected sub-second).

## Machine provenance

- Discoverer: Recipe exhaustive Cartesian traversal
- Failure fingerprint: `804153c581a2a5f9`
- Classifier provider: `openrouter`
- Classifier model: `thinkingmachines/inkling-small:free`
- Classifier effort: `default`

### Classifier decision

```json
{"verdict":"new","issue":0,"title":"NVIDIA nv0 training stalls beyond 60s on quantized layer/pool model with integer arithmetic","body":"Public `recipe.train().optimizer(adamw).lr(0.0001).seed(46945512480253).epochs(1).int(8).run()` on `backend:nv0` exceeds the 60-second threshold for one epoch on `data/numeric/sample_subfolders` using `recipe.model().layer(4).elu().norm(batch).iq(1).m .pool(2).exp().norm(batch).iq(1).m .loss(huber)`. The earliest failure is the nv0 quantized native-artifact stall during the first epoch, triggered by `.int(8)` integer arithmetic combined with batch-normalized quantized layer and pool nodes; no existing nv0 issue covers this layer/pool path (residual/attention/GRU issues do not match). Measured 60s is materially slower than the small-row numeric workload justifies (expected sub-second).","rationale":"Inspected source shows `.int(8)` drives quantized `QuantIntOp`/native compilation on nv0, and the reproduction reaches the same stall at the first training epoch through the layer-then-pool quantized path. Existing nv0 stall issues (271, 308, 291, 302, 282) specify residual, pool/attention, pool/GRU, perc/LSTM, or svm/kmeans architectures, none of which include this layer+pool with batch-normalized quantization. Thus the earliest cause is unreported."}
```

## Deterministic failure packet

kind=performance
id=804153c581a2a5f9
base=commit=5306c211d9fc2b0186a4eaf6d9e25a10caec49ca tracked_tree=modified
cursor=seed:0 cursor:10127 next:10128 composition:46945512480253
expected=one training epoch uses throughput proportionate to its operations, data volume, memory traffic, and available hardware
observed=the public Recipe run remained active after 60 seconds
measurement=elapsed_seconds:60 backend:nv0
replay=the configured slow-runtime threshold was crossed stable:true
command=cargo run --bin recipe -- /tmp/recipe-composition-repro-nv0-10127-386.rs
reproduction:
```rust
use recipe::*;

fn main() {
	let bundle = format!("/tmp/recipe-composition-repro-{}.ogdl", std::process::id());
	let data = recipe.data("data/numeric/sample_subfolders")
		.target("target")
		.norm(z_score);
	let model = recipe.model()
		.layer(4).elu().norm(batch).iq(1).m
		.pool(2).exp().norm(batch).iq(1).m
		.loss(huber);
	let report = recipe.train()
		.optimizer(adamw)
		.lr(0.0001)
		.seed(46945512480253)
		.epochs(1)
		.log(all)
		.stop(0.0)
		.int(8)
		.save(&bundle)
		.run(&model, &data);
	assert!(report.final_loss().is_finite());
}
```
backend=nv0

**Comment by nm-z** [comment]
Closing as resolved based on zero-change verification PR #424.

That PR changes 0 files and verifies the existing `minimal` behavior, so there is no code to merge. The original verification is copied below.

<details>
<summary>Verification from PR #424: Verify current NVIDIA layer and pool execution</summary>

Closes #314

This is a verification-only PR. It intentionally changes no tracked files.

## Current public result

Commit: `9098de8711552004bb9eabd84c6d8d9c120d4c58`

Command:

```text
RECIPE_DEBUG=1 ./target/release/recipe --device benji:nv0 /home/nate/.cache/recipe/issues/314/reproduction.rs
```

The exact issue model, data path, training settings, and integer arithmetic complete successfully from a cold native artifact:

- The artifact begins with `cache=miss`.
- Clang emits NVIDIA LLVM bitcode, and `llc` emits PTX.
- The artifact loads the forward, epoch, and model-load entry points.
- The 64-row epoch completes in 99.359 ms.
- Final Huber loss is finite at 1.3047.
- The process exits 0 after 6.38 seconds.

The reported 60-second NVIDIA native-artifact stall does not reproduce on current `minimal`. No production-code change is supported by this result.


</details>

Accepted verdict: PASS

Reason: The verification above reports that current `minimal` completes this issue's reported path successfully.
